src/topotoolbox/stream_object.py

```diff
@@ -385,6 +388,26 @@ def xy(self, data=None):
                         stack.append(v)
 
         return segments
+    
+    def to_shapefile(self, saving_path, saving_name):
```

Comment:
You can combine the `saving_path` and `saving_name` arguments into a single string argument. See, for example, the `write_tif` function: https://github.com/TopoToolbox/pytopotoolbox/blob/5dbc1400f37dc64c6f89b60504cfe4607c962fe2/src/topotoolbox/utils.py#L29

Then you will not need the `os` import either.